File: ScriptGraph/TemplateScriptNode/Clamp/CGClamp.lua
local CGClamp = CGClamp or {}
CGClamp.__index = CGClamp

function CGClamp.new()
    local self = setmetatable({}, CGClamp)
    self.inputs = {}
    return self
end

function CGClamp:setInput(index, func)
    self.inputs[index] = func
end

function CGClamp:getOutput(index)
    local v = self.inputs[0]()
    local minv = self.inputs[1]()
    local maxv = self.inputs[2]()
    if (v < minv) then
        return minv
    elseif (v > maxv) then
        return maxv
    else
        return v 
    end
end

return CGClamp

File: ScriptGraph/TemplateScriptNode/Clamp/CGClamp.js
/**
 * @file CGClamp.js
 * @author liujiacheng
 * @date 2021/8/23
 * @brief CGClamp.js
 * @copyright Copyright (c) 2021, ByteDance Inc, All Rights Reserved
 */

const {BaseNode} = require('./BaseNode');
const Amaz = effect.Amaz;

class CGClamp extends BaseNode {
  constructor() {
    super();
  }

  setNext(index, func) {
    this.nexts[index] = func;
  }

  setInput(index, func) {
    this.inputs[index] = func;
  }

  getOutput() {
    let inputVal = this.inputs[0]();
    let minVal = this.inputs[1]();
    let maxVal = this.inputs[2]();

    if (inputVal < minVal) {
      return minVal;
    } else if (inputVal > maxVal) {
      return maxVal;
    } else {
      return inputVal;
    }
  }
}

exports.CGClamp = CGClamp;
